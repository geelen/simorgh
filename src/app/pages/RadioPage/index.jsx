import React from 'react';
import compose from 'ramda/src/compose';
import RadioPageMain from '../../containers/RadioPageMain';

import withContexts from '../../containers/PageHandlers/withContexts';
import withPageWrapper from '../../containers/PageHandlers/withPageWrapper';
import withError from '../../containers/PageHandlers/withError';
import withLoading from '../../containers/PageHandlers/withLoading';
import withData from '../../containers/PageHandlers/withData';

const RadioContainer = props => {
  return <RadioPageMain {...props} />;
};

const EnhancedRadioContainer = compose(
  withContexts,
  withPageWrapper,
  withLoading,
  withError,
  withData,
)(RadioContainer);

export default EnhancedRadioContainer;
