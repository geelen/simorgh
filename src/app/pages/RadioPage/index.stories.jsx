import React from 'react';
import { BrowserRouter } from 'react-router-dom';
import { storiesOf } from '@storybook/react';
import { withKnobs } from '@storybook/addon-knobs';
import { withServicesKnob } from '@bbc/psammead-storybook-helpers';
import RadioPage from '.';
import indonesia from '#data/indonesia/bbc_indonesian_radio/liveradio.json';
import korean from '#data/korean/bbc_korean_radio/liveradio.json';
import tigrinya from '#data/tigrinya/bbc_tigrinya_radio/liveradio.json';
import afaanoromoo from '#data/afaanoromoo/bbc_afaanoromoo_radio/liveradio.json';
import amharic from '#data/amharic/bbc_amharic_radio/liveradio.json';
import WithTimeMachine from '#testHelpers/withTimeMachine';

const liveRadioFixtures = {
  indonesia,
  korean,
  tigrinya,
  afaanoromoo,
  amharic,
};

const matchFixtures = service => ({
  params: {
    mediaId: 'liveradio',
    serviceId: {
      indonesia: 'bbc_indonesian_radio',
      korean: 'bbc_korean_radio',
      tigrinya: 'bbc_tigrinya_radio',
      afaanoromoo: 'bbc_afaanoromoo_radio',
      amharic: 'bbc_amharic_radio',
    }[service],
  },
});

const status = 200;

storiesOf('Pages|Radio Page', module)
  .addDecorator(story => <WithTimeMachine>{story()}</WithTimeMachine>)
  .addDecorator(withKnobs)
  .addDecorator(
    withServicesKnob({
      defaultService: 'indonesia',
      services: Object.keys(liveRadioFixtures),
    }),
  )
  .add('default', ({ service }) => (
    <BrowserRouter>
      <RadioPage
        match={matchFixtures(service)}
        pageData={liveRadioFixtures[service]}
        status={status}
        service={service}
        isAmp={false}
        loading={false}
        error=""
        pageType="media"
      />
    </BrowserRouter>
  ));
