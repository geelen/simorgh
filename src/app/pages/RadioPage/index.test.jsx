import React from 'react';
import { shouldMatchSnapshot } from '@bbc/psammead-test-helpers';
import RadioPage from '.';
import amharicPageData from '#data/amharic/bbc_amharic_radio/liveradio';

const liveRadioScaffoldProps = {
  isAmp: false,
  pageType: 'media',
  service: 'amharic',
  pathname: '/pathname',
  match: {
    params: {
      serviceId: 'bbc_amharic_radio',
      mediaId: 'liveradio',
      pageData: amharicPageData,
    },
  },
  status: 200,
};

jest.mock('../../containers/PageHandlers/withPageWrapper', () => Component => {
  const PageWrapperContainer = props => (
    <div id="PageWrapperContainer">
      <Component {...props} />
    </div>
  );

  return PageWrapperContainer;
});

jest.mock('../../containers/PageHandlers/withLoading', () => Component => {
  const LoadingContainer = props => (
    <div id="LoadingContainer">
      <Component {...props} />
    </div>
  );

  return LoadingContainer;
});

jest.mock('../../containers/PageHandlers/withError', () => Component => {
  const ErrorContainer = props => (
    <div id="ErrorContainer">
      <Component {...props} />
    </div>
  );

  return ErrorContainer;
});

jest.mock('../../containers/PageHandlers/withData', () => Component => {
  const DataContainer = props => (
    <div id="DataContainer">
      <Component {...props} />
    </div>
  );

  return DataContainer;
});

jest.mock('../../containers/RadioPageMain', () => {
  const RadioPageMain = () => <div>RadioPageMain</div>;

  return RadioPageMain;
});

describe('Radio Page', () => {
  describe('snapshots', () => {
    shouldMatchSnapshot(
      'should match scaffold snapshot',
      <RadioPage {...liveRadioScaffoldProps} />,
    );
  });
});
