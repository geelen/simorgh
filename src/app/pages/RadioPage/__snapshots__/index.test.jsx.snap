// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Radio Page snapshots should match scaffold snapshot 1`] = `
<div
  id="PageWrapperContainer"
>
  <div
    id="LoadingContainer"
  >
    <div
      id="ErrorContainer"
    >
      <div
        id="DataContainer"
      >
        <div>
          RadioPageMain
        </div>
      </div>
    </div>
  </div>
</div>
`;
